consensus/src/data_manager.rs

```diff
@@ -4,53 +4,87 @@
 use std::sync::Arc;
 // use futures::channel::{mpsc, mpsc::Sender, oneshot};
 use crate::quorum_store::batch_reader::BatchReader;
+use aptos_crypto::HashValue;
 use aptos_types::transaction::SignedTransaction;
 use arc_swap::ArcSwapOption;
 use consensus_types::common::Payload;
 use consensus_types::proof_of_store::LogicalTime;
+use consensus_types::request_response::ConsensusRequest;
 use executor_types::Error;
+use futures::channel::mpsc::Sender;
 use tokio::sync::oneshot;
 
 /// Notification of execution committed logical time for QuorumStore to clean.
 #[async_trait::async_trait]
 pub trait DataManager: Send + Sync {
     /// Notification of committed logical time
-    async fn notify_commit(&self, logical_time: LogicalTime);
+    async fn notify_commit(&self, logical_time: LogicalTime, payloads: Vec<Payload>);
 
-    fn new_epoch(&self, data_reader: Arc<BatchReader>);
+    fn new_epoch(
+        &self,
+        data_reader: Arc<BatchReader>,
+        quorum_store_wrapper_tx: Sender<ConsensusRequest>,
+    );
 
-    async fn get_data(&self, payload: Payload) -> Result<Vec<SignedTransaction>, Error>;
+    async fn get_data(&self, maybe_payload: Payload) -> Result<Vec<SignedTransaction>, Error>;
```

Comment:
`maybe_payload` as part of the interface sounds weird.

Reply:
Agree.
It was an option at some point, but not anymore...